src/state/dialogs.ts

```diff
@@ -23,31 +18,28 @@ export const dialogsSlice = createSlice({
   name: 'Dialogs',
   initialState,
   reducers: {
-    openDialog: (state: DialogsState, action: PayloadAction<DialogInfo<DialogTypes>>) => {
-      if (state.activeDialog?.type === action.payload.type) return;
+    openDialog: (state, action: PayloadAction<DialogType>) => {
+      if (state.activeDialog?.tag === action.payload.tag) return;
 
       if (state.activeDialog) {
         state.dialogQueue.push(action.payload);
       } else {
         state.activeDialog = action.payload;
       }
     },
-    closeDialog: (state: DialogsState) => {
+    closeDialog: (state) => {
```

Comment:
we can let compiler infer this - truth is it isn't exactly DialogsState, it's WritableSomething<DialogsState> which isn't playing nice with some values